mantidproject/mantid
# Changes to HRPD to match up better with old scripts

**Description of work.**
The old HPRD scripts produced a differing output to the new ones, these changes allow applying Solid Angle Corrections and to set whether or not to subtract the empty instrument run, these allow closer to the original scripts outputs.
**To test:**
*Requires access to the isis data archive*
download the following [files](https://github.com/mantidproject/mantid/files/3183135/hrpd.test.zip) and update the config.yaml paths, then run the following script.
```
from isis_powder import hrpd

config_file_path = r"/path/to//hrpd_config_example.yaml"
hrpd_example = hrpd.HRPD(config_file=config_file_path,user_name="test",do_solid_angle_corrections=False)
hrpd_example.create_vanadium(first_cycle_run_no="66027", window="30-130")
hrpd_example.focus(run_number="66055", window="30-130",vanadium_normalisation = True,subtract_empty_instrument=True)
```
Then change the value of `do_solid_angle_corrections` to True and try again, then change the value of `subtract_empty_instrument` to False and try again, all three should run through fine, and the output workspaces ("66055.raw-ResultTOF_" 1,2 and,3) should be slightly different each time.

Re #23731
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
